### DISCUSSION TITLE
How to Perform Grain Reconstruction in Strained AA1100 Aluminum Alloy

### DISCUSSION THREAD

**Discussion by ejivul35p** [mtex-toolbox/mtex#2033] [Ask Anything]
Due to processing, the aluminum alloy contains numerous dislocations, resulting in small grain sizes. In areas with dislocations, the grain reconstruction appears unusual. This could be due to an inability to identify the corresponding grain boundary angles or other contributing factors. I'd like to inquire about methods to effectively reconstruct grains in dislocated regions. Is there a way to reconstruct grains in these dislocated areas? Alternatively, is it possible to exclude the dislocated portions while performing grain reconstruction and focus solely on calculating the grain sections?

```matlab
% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','csv');

plot(ebsd)


grains = calcGrains(ebsd,'angle',1.5*degree)

% start overide mode


% plot the boundary of all grains
figure(2);
plot(grains.boundary,'linewidth',1)

% stop overide mode
%hold off
%close all; plotx2east

% import the data
%mtexdata ferrite silent

% grain reconstruction
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'), 'angle', 1.5*degree);

% remove small grains
ebsd(grains(grains.grainSize < 2)) = [];

% reidentify grains with small grains removed:
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',1.5*degree);

grains = smooth(grains,5);
plot(grains.boundary,'lineWidth',0.3);


% plot the data
plot(ebsd, ebsd.orientations)
```

This is an EBSD map of aluminum.
![300_20_85 IQ Bar 58](https://github.com/mtex-toolbox/mtex/assets/156154684/23f7077b-e095-47f7-8534-1a82ac191f78)

This is an image showing grain reconstruction.
![300_20_85-1 5-3 bar](https://github.com/mtex-toolbox/mtex/assets/156154684/63861196-30a6-47be-9c7e-223bc87b7eda)

**Comment by lfgmorales** [comment]
Hi

First you may want to reconsider your threshold angle for the definition of “grain”, you are usinf 1.5 degree, which is certainly too low and include low angle grain boundaries. Have a look here how to compute low angle grain boundaries together with grains as individual entities [Subgrain Boundaries](https://mtex-toolbox.github.io/SubGrainBoundaries.html)

Second, consider removing grains with a larger number of pixels, at the moment you only remove grains smaller than 2 pixels, which is a rather low value

Also it seems from your band contrast map ( or
Image quality) the step size you have used in this map is not ideal for the grain size of your microstructure, but this is an educated guess

Cheers

Luiz

**Reply by ejivul35p** [reply]
Thank you, this is very helpful. I have another question: If I want to calculate grain sizes, will subgrains be included in the calculation? I am interested in knowing the size of subgrains.

**Comment by lfgmorales** [comment]
Hi

It all depends on your definition of “grains” as individual entities, particularly related to the misorientation chosen for the grains calculation.

Cheers

Luiz


Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 19 Jan 2024, at 07:49, ejivul35p ***@***.***> wrote:

﻿

Thank you, this is very helpful. I have another question: If I want to calculate grain sizes, will subgrains be included in the calculation? I am interested in knowing the size of subgrains.

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2033#discussioncomment-8177859>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5INZVBUFITN4IA3Y4UX3YPIJN7AVCNFSM6AAAAABBUDCSU2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DCNZXHA2TS>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2033#discussioncomment-8177859", "url": "https://github.com/mtex-toolbox/mtex/discussions/2033#discussioncomment-8177859", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]
